plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java

```diff
@@ -203,6 +203,9 @@ public void afterRunProcess(String output, String errors, IExternalCodeAnalysisS
                     if (severityFound.equals("warning")) {
                         markerSeverity = IMarker.SEVERITY_WARNING;
                     }
+                    if (severityFound.equals("note")) {
```

Comment:
This is not the proper fix as you're effectively creating a different marker for each note.

What needs to be done is that when you find an error or warning, instead of creating it right away you need to keep it in memory and later, when a note is found in the same line/column you have to add the note message to the same place.

So, you need some structure to map the line/colto the message information (message, region) and then when there's a note in the same line/col just amend the message instead of creating a new marker.

Something as: `Map<LineCol, MessageInfo> lineColToMessage = new HashMap<>()` (note, you have to create those messages and the LineCol must have equals/hashCode overridden.

Hint: you can create something as:

```
    private static class LineCol{
        private final int line;
        private final int col;
    }
```

Then select the `LineCol` and do `Alt+Shift+S to auto-generate the constructor and equals()/hashCode().

Then, after the processing is done actually create the markers based on the full message.

Also, you may need to take care of having 2 different errors in the same line/message (so, if you find an error with notes, when a new error comes along you should not override the previous, rather, create the marker for it right away and only then store the new error to receive further notes).

p.s.: Don't forget to create a test ;)